ext/standard/tests/math/round_bug12143.phpt

```diff
@@ -0,0 +1,22 @@
+--TEST--
+Bug #12143 round() with large FP
+--FILE--
+<?php
+var_dump(round(0.49999999999999994, 0, PHP_ROUND_HALF_UP) == 0);
+var_dump(round(0.49999999999999994, 0, PHP_ROUND_HALF_DOWN) == 0);
+var_dump(round(0.49999999999999994, 0, PHP_ROUND_HALF_EVEN) == 0);
+var_dump(round(0.49999999999999994, 0, PHP_ROUND_HALF_ODD) == 0);
+var_dump(round(-0.49999999999999994, 0, PHP_ROUND_HALF_UP) == -0);
+var_dump(round(-0.49999999999999994, 0, PHP_ROUND_HALF_DOWN) == 0);
+var_dump(round(-0.49999999999999994, 0, PHP_ROUND_HALF_EVEN) == 0);
+var_dump(round(-0.49999999999999994, 0, PHP_ROUND_HALF_ODD) == 0);
```

Comment:
It might be better to dump the resulting number, rather than the result of the comparison.

Reply:
Changed it. And also added cases for the number `0.5000000000000004`.